### PR TITLE
Add more img2img Pipelines

### PR DESCRIPTION
**Pull request by radames**
This is an attempt to add a few more im2img pipelines.

This PR adds support for `StableDiffusionImageVariationPipeline`, `StableUnCLIPImg2ImgPipeline` and finally `FlaxStableDiffusionControlNetPipeline` 

Two points of concern:
1. I'm using Diffusers 0.17.0.dev0 - git+https://github.com/huggingface/diffusers@main since it has support for the new`FlaxStableDiffusionControlNetPipeline`, and PyTorch 2.0
2. Running Flax/Jax on a GPU container was a bit challenging, I'm not sure if this will be compatible with our API backend, currently to make it work on my test enviroment I needed to use `nvidia/cuda:11.7.1-devel-ubuntu22.04` and `jax[cuda11_cudnn82]` the models take a considerable time to load, then the inference time was reasonable, using `Tesla V100 16VRAM` @Narsil wdyt? 

If the Flax is to complicated I can remove it from this PR. thanks

You can run tests on your env and pick a model and pipeline.

Install requirements, you also need docker/GPU

```pip install -r requirements.txt # for pytests, quality and manager``` 
```pip install -r docker_images/diffusers/requirements.txt # for local pipeline requirements```
Run all tests `pytest -sv --rootdir docker_images/diffusers docker_images/diffusers` 

Run a single model via Docker, then you can send requests
```DEBUG=1 ./manage.py docker "mfidabel/controlnet-segment-anything" --gpu```
or local
```DEBUG=1 ./manage.py start "mfidabel/controlnet-segment-anything" --gpu```
then you can test via https://observablehq.com/d/6ff4df5014d1ed1f

```python
TESTABLE_MODELS: Dict[str, List[str]] = {
    "text-to-image": ["hf-internal-testing/tiny-stable-diffusion-pipe-no-safety"],
    "image-to-image": [
        "hf-internal-testing/tiny-controlnet",
        "hf-internal-testing/tiny-stable-diffusion-pix2pix",
        # "radames/stable-diffusion-v1-5-img2img",  # stable diffusion
        # "radames/instruct-pix2pix-img2img",  # instruct pix2pix
        # "lllyasviel/sd-controlnet-depth",  # controlnet
        # "lllyasviel/control_v11f1e_sd15_tile",  # controlnet
        # "radames/AltDiffusion-m9-img2img",  # alt diffusion
        # "lambdalabs/sd-image-variations-diffusers",  # StableDiffusionImageVariationPipeline
        # "radames/stable-diffusion-2-1-unclip-small-img2img",  # StableDiffusionImg2ImgPipeline
        # # "stabilityai/stable-diffusion-2-1-unclip",  # StableDiffusionImg2ImgPipeline
        # "mfidabel/controlnet-segment-anything",  # flax controlnet
        #”Vincent-luo/controlnet-hands", # flax controlnet
    ],
}
```

### CODE REVIEW

File: docker_images/diffusers/app/pipelines/image_to_image.py [hunk 1]
```diff
@@ -58,34 +76,66 @@ def __init__(self, model_id: str):
...
                 model_id, use_auth_token=use_auth_token, **kwargs
             )
 
-        if torch.cuda.is_available():
+        if torch.cuda.is_available() and not isinstance(
+            self.ldm, FlaxStableDiffusionControlNetPipeline
+        ):
             self.ldm.to("cuda")
-            self.ldm.enable_xformers_memory_efficient_attention()
+            self.ldm.unet.set_attn_processor(AttnProcessor2_0())
```
Comment (patrickvonplaten) [inline docker_images/diffusers/app/pipelines/image_to_image.py:125]: Nice!

**Review by patrickvonplaten — COMMENTED** [review]
Ok for me! I wonder whether it already makes sense to have Flax API support because we can easily convert every Flax model to PyTorch and the run inference API on the PyTorch weights. 

Having both Flax and PyTorch here might bloat the code a bit?

@radames any flax models in particular that you would like to see working?

File: docker_images/diffusers/requirements.txt [hunk 2]
```diff
@@ -2,11 +2,18 @@ starlette==0.25.0
 api-inference-community==0.0.30
 huggingface_hub==0.14.1
 safetensors==0.3.1
-diffusers==0.16.1
+# diffusers==0.16.1
+git+https://github.com/huggingface/diffusers@main
```
Comment (Narsil) [inline docker_images/diffusers/requirements.txt:6]: Any reason to use main ? ( It works, but it's harder to know when something was built, leading to confusion when some features is missing from here, while it is actually on main)

**Review by Narsil — APPROVED** [review]
LGTM.

File: docker_images/diffusers/requirements.txt (hunk 2 above)
Comment (radames) [reply docker_images/diffusers/requirements.txt:6]: you're right!, @patrickvonplaten what would be a good commit to pin here? I can update when 0.17 come out

**Comment by radames** [comment]
> Having both Flax and PyTorch here might bloat the code a bit?
> 
> @radames any flax models in particular that you would like to see working?

Completely agree, I'll remove the flax part, add a warning saying Flax is not compatible please convert to PyTorch. I thought you wanted to support the new Flax models that come out from the jax-diffusers-event. I have no particular model in mind, but it would be nice to covert the ones on the [leaderboard](https://huggingface.co/spaces/jax-diffusers-event/leaderboard).

File: docker_images/diffusers/requirements.txt (hunk 2 above)
Comment (Narsil) [reply docker_images/diffusers/requirements.txt:6]: Commit is the best in between !

**Comment by radames** [comment]
removed the flax @patrickvonplaten, thanks

**Review by patrickvonplaten — APPROVED** [review]
Great!

**Pull request merged**